sql/core/src/test/scala/org/apache/spark/sql/test/DataFrameReaderWriterSuite.scala

```diff
@@ -1219,4 +1219,15 @@ class DataFrameReaderWriterSuite extends QueryTest with SharedSparkSession with
       }
     }
   }
+
+  test("forbid char like data types in reader schema api") {
+    val ddls = Seq("a char(1)", "b varchar(10)", "c array<char(5)>")
+    ddls.foreach { dt =>
+      val e = intercept[AnalysisException](spark.read.schema(dt))
```

Comment:
isn't it covered by https://github.com/apache/spark/pull/30586/files#diff-928d574dedbca234036aa1f8b0098efdf914b74d2d6071cb439ab3d2025a25c3R463 ?